### ISSUE TITLE
`master` - Field Filter with category can cause ambiguous column reference

### ISSUE THREAD

**Issue by flamber** [metabase/metabase#15460]
**Describe the bug**
On `master` - not possible to use Field Filter (Category, any of them) on a query with joins, where the tables have similar columns.
Doing something similar with the Field Filter of a date type does not cause this problem, since it correctly references the full path.

**To Reproduce**
1. Native query > Sample Dataset - make `{{category}}` a Field Filter referencing Products.Category
```
select p.created_at, products.category
from products
left join products p on p.id=products.id
where {{category}}
```
2. Input "Doohickey" in the filter widget and run the query
![image](https://user-images.githubusercontent.com/1447303/113433099-d9247500-93de-11eb-8443-4f859f048c0f.png)
3. This fails with error `Ambiguous column name "CATEGORY";` because the replaced variable only has the column and not the full schema+table+column as 0.38.3 uses.

<details><summary>Full stacktrace</summary>

```
[78edc18e-2bec-42b8-907c-0d2d8a4f0263] 2021-04-02T18:13:27+02:00 ERROR metabase.query-processor.middleware.catch-exceptions Error processing query: null
{:database_id 1,
 :started_at #t "2021-04-02T16:13:27.264985Z[GMT]",
 :state "90059",
 :json_query
 {:type "native",
  :native
  {:query
   "select p.created_at, products.category\nfrom products\nleft join products p on p.id=products.id\nwhere {{category}}",
   :template-tags
   {:category
    {:id "c16861a6-1394-66b6-3a24-c4ccf7585242",
     :name "category",
     :display-name "Category",
     :type "dimension",
     :dimension ["field" 16 nil],
     :widget-type "category/=",
     :default nil}}},
  :database 1,
  :parameters [{:type "string/=", :value ["Doohickey"], :target ["dimension" ["template-tag" "category"]]}],
  :middleware {:js-int-to-string? true, :add-default-userland-constraints? true}},
 :status :failed,
 :class org.h2.jdbc.JdbcSQLException,
 :stacktrace
 ["org.h2.message.DbException.getJdbcSQLException(DbException.java:357)"
  "org.h2.message.DbException.get(DbException.java:179)"
  "org.h2.message.DbException.get(DbException.java:155)"
  "org.h2.expression.ExpressionColumn.mapColumn(ExpressionColumn.java:127)"
  "org.h2.expression.ExpressionColumn.mapColumns(ExpressionColumn.java:97)"
  "org.h2.expression.Comparison.mapColumns(Comparison.java:510)"
  "org.h2.command.dml.Select.mapColumns(Select.java:1240)"
  "org.h2.command.dml.Select.init(Select.java:826)"
  "org.h2.command.Parser.parseSelect(Parser.java:1925)"
  "org.h2.command.Parser.parsePrepared(Parser.java:463)"
  "org.h2.command.Parser.parse(Parser.java:335)"
  "org.h2.command.Parser.parse(Parser.java:307)"
  "org.h2.command.Parser.prepareCommand(Parser.java:278)"
  "org.h2.engine.Session.prepareLocal(Session.java:611)"
  "org.h2.engine.Session.prepareCommand(Session.java:549)"
  "org.h2.jdbc.JdbcConnection.prepareCommand(JdbcConnection.java:1247)"
  "org.h2.jdbc.JdbcPreparedStatement.<init>(JdbcPreparedStatement.java:76)"
  "org.h2.jdbc.JdbcConnection.prepareStatement(JdbcConnection.java:1135)"
  "com.mchange.v2.c3p0.impl.NewProxyConnection.prepareStatement(NewProxyConnection.java:267)"
  "--> driver.sql_jdbc.execute$fn__78926.invokeStatic(execute.clj:262)"
  "driver.sql_jdbc.execute$fn__78926.invoke(execute.clj:260)"
  "driver.sql_jdbc.execute$prepared_statement_STAR_.invokeStatic(execute.clj:302)"
  "driver.sql_jdbc.execute$prepared_statement_STAR_.invoke(execute.clj:299)"
  "driver.sql_jdbc.execute$statement_or_prepared_statement.invokeStatic(execute.clj:326)"
  "driver.sql_jdbc.execute$statement_or_prepared_statement.invoke(execute.clj:323)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:466)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:453)"
  "driver.sql_jdbc.execute$execute_reducible_query.invokeStatic(execute.clj:462)"
  "driver.sql_jdbc.execute$execute_reducible_query.invoke(execute.clj:453)"
  "driver.sql_jdbc$fn__80600.invokeStatic(sql_jdbc.clj:54)"
  "driver.sql_jdbc$fn__80600.invoke(sql_jdbc.clj:52)"
  "driver.h2$fn__79389.invokeStatic(h2.clj:83)"
  "driver.h2$fn__79389.invoke(h2.clj:80)"
  "query_processor.context$executef.invokeStatic(context.clj:59)"
  "query_processor.context$executef.invoke(context.clj:48)"
  "query_processor.context.default$default_runf.invokeStatic(default.clj:68)"
  "query_processor.context.default$default_runf.invoke(default.clj:66)"
  "query_processor.context$runf.invokeStatic(context.clj:45)"
  "query_processor.context$runf.invoke(context.clj:39)"
  "query_processor.reducible$pivot.invokeStatic(reducible.clj:34)"
  "query_processor.reducible$pivot.invoke(reducible.clj:31)"
  "query_processor.middleware.mbql_to_native$mbql__GT_native$fn__47111.invoke(mbql_to_native.clj:25)"
  "query_processor.middleware.check_features$check_features$fn__46353.invoke(check_features.clj:39)"
  "query_processor.middleware.limit$limit$fn__47097.invoke(limit.clj:37)"
  "query_processor.middleware.cache$maybe_return_cached_results$fn__46002.invoke(cache.clj:211)"
  "query_processor.middleware.optimize_temporal_filters$optimize_temporal_filters$fn__47357.invoke(optimize_temporal_filters.clj:204)"
  "query_processor.middleware.validate_temporal_bucketing$validate_temporal_bucketing$fn__49072.invoke(validate_temporal_bucketing.clj:50)"
  "query_processor.middleware.auto_parse_filter_values$auto_parse_filter_values$fn__45121.invoke(auto_parse_filter_values.clj:43)"
  "query_processor.middleware.wrap_value_literals$wrap_value_literals$fn__41491.invoke(wrap_value_literals.clj:161)"
  "query_processor.middleware.annotate$add_column_info$fn__41366.invoke(annotate.clj:598)"
  "query_processor.middleware.permissions$check_query_permissions$fn__46225.invoke(permissions.clj:81)"
  "query_processor.middleware.pre_alias_aggregations$pre_alias_aggregations$fn__48000.invoke(pre_alias_aggregations.clj:40)"
  "query_processor.middleware.cumulative_aggregations$handle_cumulative_aggregations$fn__46426.invoke(cumulative_aggregations.clj:60)"
  "query_processor.middleware.resolve_joined_fields$resolve_joined_fields$fn__48299.invoke(resolve_joined_fields.clj:102)"
  "query_processor.middleware.resolve_joins$resolve_joins$fn__48612.invoke(resolve_joins.clj:171)"
  "query_processor.middleware.add_implicit_joins$add_implicit_joins$fn__44701.invoke(add_implicit_joins.clj:190)"
  "query_processor.middleware.large_int_id$convert_id_to_string$fn__47061.invoke(large_int_id.clj:59)"
  "query_processor.middleware.format_rows$format_rows$fn__47042.invoke(format_rows.clj:74)"
  "query_processor.middleware.desugar$desugar$fn__46492.invoke(desugar.clj:21)"
  "query_processor.middleware.binning$update_binning_strategy$fn__45508.invoke(binning.clj:227)"
  "query_processor.middleware.resolve_fields$resolve_fields$fn__46028.invoke(resolve_fields.clj:34)"
  "query_processor.middleware.add_dimension_projections$add_remapping$fn__44350.invoke(add_dimension_projections.clj:312)"
  "query_processor.middleware.add_implicit_clauses$add_implicit_clauses$fn__44579.invoke(add_implicit_clauses.clj:147)"
  "query_processor.middleware.upgrade_field_literals$upgrade_field_literals$fn__49021.invoke(upgrade_field_literals.clj:40)"
  "query_processor.middleware.add_source_metadata$add_source_metadata_for_source_queries$fn__44864.invoke(add_source_metadata.clj:123)"
  "query_processor.middleware.reconcile_breakout_and_order_by_bucketing$reconcile_breakout_and_order_by_bucketing$fn__48174.invoke(reconcile_breakout_and_order_by_bucketing.clj:100)"
  "query_processor.middleware.auto_bucket_datetimes$auto_bucket_datetimes$fn__45068.invoke(auto_bucket_datetimes.clj:133)"
  "query_processor.middleware.resolve_source_table$resolve_source_tables$fn__46075.invoke(resolve_source_table.clj:45)"
  "query_processor.middleware.parameters$substitute_parameters$fn__47982.invoke(parameters.clj:111)"
  "query_processor.middleware.resolve_referenced$resolve_referenced_card_resources$fn__46127.invoke(resolve_referenced.clj:79)"
  "query_processor.middleware.expand_macros$expand_macros$fn__46748.invoke(expand_macros.clj:155)"
  "query_processor.middleware.add_timezone_info$add_timezone_info$fn__44873.invoke(add_timezone_info.clj:15)"
  "query_processor.middleware.splice_params_in_response$splice_params_in_response$fn__48974.invoke(splice_params_in_response.clj:32)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48185$fn__48189.invoke(resolve_database_and_driver.clj:31)"
  "driver$do_with_driver.invokeStatic(driver.clj:60)"
  "driver$do_with_driver.invoke(driver.clj:56)"
  "query_processor.middleware.resolve_database_and_driver$resolve_database_and_driver$fn__48185.invoke(resolve_database_and_driver.clj:25)"
  "query_processor.middleware.fetch_source_query$resolve_card_id_source_tables$fn__46988.invoke(fetch_source_query.clj:274)"
  "query_processor.middleware.store$initialize_store$fn__48983$fn__48984.invoke(store.clj:11)"
  "query_processor.store$do_with_store.invokeStatic(store.clj:44)"
  "query_processor.store$do_with_store.invoke(store.clj:38)"
  "query_processor.middleware.store$initialize_store$fn__48983.invoke(store.clj:10)"
  "query_processor.middleware.validate$validate_query$fn__49028.invoke(validate.clj:10)"
  "query_processor.middleware.normalize_query$normalize$fn__47124.invoke(normalize_query.clj:22)"
  "query_processor.middleware.add_rows_truncated$add_rows_truncated$fn__44719.invoke(add_rows_truncated.clj:35)"
  "query_processor.middleware.results_metadata$record_and_return_metadata_BANG_$fn__48959.invoke(results_metadata.clj:147)"
  "query_processor.middleware.constraints$add_default_userland_constraints$fn__46369.invoke(constraints.clj:42)"
  "query_processor.middleware.process_userland_query$process_userland_query$fn__48071.invoke(process_userland_query.clj:135)"
  "query_processor.middleware.catch_exceptions$catch_exceptions$fn__46309.invoke(catch_exceptions.clj:173)"
  "query_processor.reducible$async_qp$qp_STAR___38069$thunk__38070.invoke(reducible.clj:103)"
  "query_processor.reducible$async_qp$qp_STAR___38069.invoke(reducible.clj:109)"
  "query_processor.reducible$sync_qp$qp_STAR___38078$fn__38081.invoke(reducible.clj:135)"
  "query_processor.reducible$sync_qp$qp_STAR___38078.invoke(reducible.clj:134)"
  "query_processor$process_userland_query.invokeStatic(query_processor.clj:239)"
  "query_processor$process_userland_query.doInvoke(query_processor.clj:235)"
  "query_processor$fn__49118$process_query_and_save_execution_BANG___49127$fn__49130.invoke(query_processor.clj:251)"
  "query_processor$fn__49118$process_query_and_save_execution_BANG___49127.invoke(query_processor.clj:243)"
  "query_processor$fn__49162$process_query_and_save_with_max_results_constraints_BANG___49171$fn__49174.invoke(query_processor.clj:263)"
  "query_processor$fn__49162$process_query_and_save_with_max_results_constraints_BANG___49171.invoke(query_processor.clj:256)"
  "api.dataset$run_query_async$fn__55336.invoke(dataset.clj:56)"
  "query_processor.streaming$streaming_response_STAR_$fn__55315$fn__55316.invoke(streaming.clj:72)"
  "query_processor.streaming$streaming_response_STAR_$fn__55315.invoke(streaming.clj:71)"
  "async.streaming_response$do_f_STAR_.invokeStatic(streaming_response.clj:65)"
  "async.streaming_response$do_f_STAR_.invoke(streaming_response.clj:63)"
  "async.streaming_response$do_f_async$fn__16061.invoke(streaming_response.clj:84)"],
 :context :ad-hoc,
 :error
 "Ambiguous column name \"CATEGORY\"; SQL statement:\n-- Metabase:: userID: 44 queryType: native queryHash: a1cc705d669e665066ad428394e4ab01393a5ce8359b57dd74c116db3b44fc63\nselect p.created_at, products.category\nfrom products\nleft join products p on p.id=products.id\nwhere CATEGORY = ? [90059-197]",
 :row_count 0,
 :running_time 0,
 :data {:rows [], :cols []}}
```

</details>

**Information about your Metabase Installation:**
master commit `dfa9398`

Reported by @ukritw, who is currently blocked by this in some questions.

**Comment by flamber** [comment]
Fixed in 0.39.0-rc1 and will merged into `master` within a few of days.
